cms/src/routes/genomicVariants.js

```diff
@@ -76,6 +79,59 @@ variantsRouter.post('/import/:name', async (req, res) => {
   }
 });
 
+variantsRouter.post('/check', async (req, res) => {
+  const { models } = req.body;
+
+  if (!Array.isArray(models) || models.length < 1) {
+    logger.error(
+      'GDC model status check failed due to a bad request. `models` must be an array of model names.',
+    );
+    return res.status(400).json({
+      error:
+        'GDC model status check failed due to a bad request. `models` must be an array of model names.',
+    });
+  }
+
+  try {
+    const results = Object.values(GDC_MODEL_STATES).reduce((o, key) => ({ ...o, [key]: [] }), {});
+    // GDC appears to have a rate limiter on GQL requests, so split into batches
+    // max. working batch size was found to be 100
+    // min. working delay between batches was found to be 500ms
+    const BATCH_SIZE = 100;
+    const BATCH_DELAY = 500;
+    const batches = _.chunk(models, BATCH_SIZE);
+
+    for (let i = 0; i < batches.length; i++) {
+      await new Promise(async (resolve, reject) => {
+        await Promise.all(
+          batches[i].map(async model => {
+            logger.debug(`Checking GDC state for model: ${model}`);
+            const modelStatus = await getMafStatus(model);
+            results[modelStatus.status].push(model);
+          }),
+        )
+          .then(_ => {
+            setTimeout(() => resolve(), batches.length > 1 ? BATCH_DELAY : 0);
```

Comment:
shouldn't we wait for a batch to complete before starting the next batch, instead of just delaying by a set amount?